src/net.cpp

```diff
@@ -812,6 +813,33 @@ static bool AttemptToEvictConnection(bool fPreferNewConnection) {
 
     // Protect connections with certain characteristics
 
+    // Check version of eviction candidates and prioritize nodes which do not support network upgrade.
+    std::vector<CNodeRef> vTmpEvictionCandidates;
+    int height;
+    {
+        LOCK(cs_main);
+        height = chainActive.Height();
+    }
+
+    const Consensus::Params& params = Params().GetConsensus();
+    int nActivationHeight = params.vUpgrades[Consensus::UPGRADE_OVERWINTER].nActivationHeight;
+
+    if ((height < nActivationHeight) &&
```

Comment:
This might need to be `height + 1 < nActivationHeight` to match the above changes.

Reply:
(or `height = chainActive.Height() + 1` further up)

Reply:
Ditto above.

Reply:
I was momentarily concerned that this might have a wrapping problem, but both `height` and `nActivationHeight` are `int`s, and `nActivationHeight == -1` for mainnet in 1.0.15, so `height < nActivationHeight` will always evaluate to false.